### DISCUSSION TITLE
How to remove getinitialprops in stages?

### DISCUSSION THREAD

**Discussion by rahul3103** [vercel/next.js#28821] [Help] [answered]
We were using next js for a long time before `getstaticProps` and `getServersideProps`. We have `getinitialprops` in `_app`, which prevents adding the other two methods of data fetching. Is there any way we can remove `getinitialprops` in stages? We are using the next `redux wrapper` config for the redux part.

**Comment by rafaelalmeidatk** [comment, marked as answer]
Using `getInitialProps` in `_app` does not prevent you from using the new data-fetching methods, it only disables the [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization). You just need to pay attention for the different behavior:

- If you add `getServerSideProps` to your page, `getInitialProps` from `_app` will always run on the server for every request for this page.
- If you add `getStaticProps` to your page, `getInitialProps` from `_app` will run at build time for this page, so you can't have custom logic based on the request there
